tokio/src/runtime/task/harness.rs

```diff
@@ -202,17 +212,13 @@ where
                     return PollFuture::Complete;
                 }
 
-                match self.state().transition_to_idle() {
-                    TransitionToIdle::Ok => PollFuture::Done,
-                    TransitionToIdle::OkNotified => PollFuture::Notified,
-                    TransitionToIdle::OkDealloc => PollFuture::Dealloc,
-                    TransitionToIdle::Cancelled => {
-                        // The transition to idle failed because the task was
-                        // cancelled during the poll.
-                        cancel_task(self.core());
-                        PollFuture::Complete
-                    }
+                let transition_res = self.state().transition_to_idle();
+                if let &TransitionToIdle::Cancelled = &transition_res {
```

Comment:
I'm pretty sure that this will compile.
```suggestion
                if let TransitionToIdle::Cancelled = transition_res {
```